### DISCUSSION TITLE
Pylance fails to resolve import when notebook is connected to a remote server

### DISCUSSION THREAD

**Discussion by CarlKilhart** [microsoft/vscode-jupyter#14025] [Questions and Answers]
Pylance works correctly when notebook is connected to a local kernel. But when I start a jupyter server with the same kernel, and connect notebook to this server, Pylance reports "Import could not be resolved from source" on some packages, such as pandas and torch. Code execution always works well. Only something wrong with the performance of Pylance in notebook. I wonder whether this is an unsupported feature, or I need to change some setting to fix it.

**Comment by DonJayamanne** [comment]
> But when I start a jupyter server with the same kernel, and connect notebook to this server,

are you connected to a local or remote kernel?

**Reply by CarlKilhart** [reply]
I'm connected to a remote kernel

**Comment by DonJayamanne** [comment]
Unfortunately this is not supported today, please create an issue on Pylance repo.

**Comment by dnszero** [comment]
Currently this does work in my testing, but only if I close and re-open the notebook after selecting a kernel.

It seems that a notebook must have a kernel selected before it is opened for Pylance to evaluate those imports properly.

I know this is just a work-around, but I hope it helps someone else.

**Comment by Thessssift** [comment]
Same problem occurred to me . When opening an already saved .ipnb Pylance works just fine but when I create a new one it doesn't seem to work . To fix this I run a new Jupiter file I want to work with save it even if it's empty , when I reopen it again then Pylance works finally .
